File: Models/Book.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookDB.Models {
    public class Book {

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int BookId { get; set; }
        public string? Title { get; set; }
        public int? PageCount { get; set; }

        public int? AuthorId { get; set; }
        public virtual Author Author { get; set; }

        public int? PublisherId { get; set; }
        public virtual Publisher Publisher { get; set; }

        public virtual ICollection<Genre> Genres { get; set; }
    }
}

File: ClientApp/src/app/books/book.ts
import { IAuthor } from "../authors/author";
import { IGenre } from "../genres/genre";
import { IPublisher } from "../publishers/publisher";

export interface IBook {
  bookId: number;
  title: string;
  pageCount: number;
  authorId: number;
  author: IAuthor;
  publisherId: number;
  publisher: IPublisher;
  genres: IGenre[];
}


